charts/elasticsearch/templates/client/es-client-deployment.yaml

```diff
@@ -115,14 +115,27 @@ spec:
           protocol: TCP
         readinessProbe:
           httpGet:
-            path: /_cluster/health
+            # local: If true, the request retrieves information from the local node only.
+            # Defaults to false, which means information is retrieved from the master node.
+            path: /_cluster/health?local=true
             port: {{ .Values.common.ports.http }}
           initialDelaySeconds: 5
         livenessProbe:
           httpGet:
             path: /_cluster/health?local=true
             port: {{ .Values.common.ports.http }}
           initialDelaySeconds: 90
+        {{- if gt (.Capabilities.KubeVersion.Minor | atoi) 15 }}
+        # This feature is available in Kubernetes 1.16+
+        startupProbe:
+          httpGet:
+            # without local=true, we are waiting for this pod's
+            # es-master to report that the cluster is ready.
+            path: /_cluster/health
+            port: {{ .Values.common.ports.http }}
+          failureThreshold: 30
+          periodSeconds: 10
```

Comment:
I'm curious about this failure threshold on the startup probe. These values state that we expect ES to take five minutes to start up. Does it actually take that long? That seems long to me, but every configuration is different. In general this should be pretty close to the real startup time. (threshold * seconds = amount of time we expect it to fail)

Reply:
I'm not sure. The reason I chose 5 minutes was because that's the default timeout for helm install. I figured we wouldn't want the service to fail, but rather to wait all the way up to its deadline? I am OK to reconfigure this though.

Reply:
We can leave it for now. I was just more curious than anything.